linked_list.py

```diff
@@ -7,32 +8,33 @@ def __init__(self, data, nextNode=None):
         self.nextNode = nextNode
 
     def __str__(self):
-        return self.data
+        return str(self.data)
 
     def __repr__(self):
-        return self.data
+        return repr(self.data)
 
 
 class LinkedList(object):
     def __init__(self, firstNode=None):
         self.firstNode = firstNode
 
-    def insert(self, newNode):
-        # insert newNode at beginning of list
+    def insert(self, val):
+        # insert val at beginning of list
+        self.newNode = Node(val)
```

Comment:
Will you ever use `newNode` outside the context of this one method?  If not, is there any reason to hook it up to the instance as an attribute like this?  You could simply bind `newNode` locally, without attaching it to `self` and achieve the same results with fewer lasting side-effects.